clang/tools/archgen-mlir/lib/LowerApprox/LowerApprox.cpp

```diff
@@ -0,0 +1,516 @@
+//===- LowerApprox.cpp - Replace approx by its approximation --------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
```

Comment:
Huge lacking comment explaining the hidden gems of this very touristic file. :-)